datarobot_provider/operators/ai_catalog.py

```diff
@@ -117,3 +118,89 @@ def execute(self, context: Dict[str, Any]) -> str:
         )
 
         return ai_catalog_dataset.version_id
+
+
+class CreateDatasetFromJDBCOperator(BaseOperator):
+    """
+    Loading dataset from JDBC Connection to DataRobot AI Catalog and return Dataset ID.
+
+    :param datarobot_conn_id: Connection ID, defaults to `datarobot_default`
+    :type datarobot_conn_id: str, optional
+    :return: DataRobot AI Catalog dataset ID
+    :rtype: str
+    """
+
+    # Specify the arguments that are allowed to parse with jinja templating
+    template_fields: Iterable[str] = []
+    template_fields_renderers: Dict[str, str] = {}
+    template_ext: Iterable[str] = ()
+    ui_color = '#f4a460'
+
+    def __init__(
+        self,
+        *,
+        datarobot_conn_id: str = "datarobot_default",
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.datarobot_conn_id = datarobot_conn_id
+        if kwargs.get('xcom_push') is not None:
+            raise AirflowException(
+                "'xcom_push' was deprecated, use 'BaseOperator.do_xcom_push' instead"
+            )
+
+    def execute(self, context: Dict[str, Any]) -> str:
+        # Initialize DataRobot client
+        DataRobotHook(datarobot_conn_id=self.datarobot_conn_id).run()
+
+        # Fetch stored JDBC Connection with credentials
+        credential_data, data_store = JDBCDataSourceHook(
+            datarobot_jdbc_conn_id=context["params"]["datarobot_jdbc_connection"]
+        ).run()
+
+        dataset_name = context["params"]["dataset_name"]
+
+        data_source = None
+
+        for dr_source_item in dr.DataSource.list():
+            if dr_source_item.canonical_name == dataset_name:
+                data_source = dr_source_item
+                self.log.info(f"Found existing DataSource:{dataset_name}, id={data_source.id}")
+                break
```

Comment:
Minor pythonic nit, but this is the perfect case for Python's `for-else` support:
```python
for dr_source_item in dr.DataSource.list():
    if dr_source_item.canonical_name == dataset_name:
        data_source = dr_source_item
        break
else:
    # No existing datasource was found so create one
    data_source = dr.DataSource.create(...)
```